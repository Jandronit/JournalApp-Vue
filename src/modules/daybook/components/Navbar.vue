<script>
import { defineComponent } from "vue";
import useAuth from "@/modules/auth/composables/useAuth";
import { useRouter } from "vue-router";
export default defineComponent({
  name: "Navbar",
  setup() {
    const router = useRouter();
    const { username, logout } = useAuth();
    return {
      username,
      onLogout: () => {
        router.push({ name: "login" });
        logout();
      },
    };
  },
});
</script>
<template>
  <nav class="navbar bg-primary">
    <a class="navbar-brand text-white" href="">
      <img
        src="@/assets/logo.png"
        alt="Vue logo"
        height="24"
        class="d-inline-block align-text-top mx-2"
      />
      {{ username }}
    </a>

    <div class="d-flex">
      <button @click="onLogout" class="btn btn-outline-light mx-2">
        <i class="fas fa-sign-out-alt"></i>
      </button>
      <button class="btn btn-outline-light mx-2">Register</button>
    </div>
  </nav>
</template>
