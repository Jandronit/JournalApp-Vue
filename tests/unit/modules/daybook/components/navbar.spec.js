import { shallowMount } from "@vue/test-utils";
import Navbar from "@/modules/daybook/components/Navbar.vue";
import createVuexStore from "../../../../mocks/mock-store";

const mockRouter = {
  push: jest.fn(),
};
describe("Tests Navbar Component", () => {
  const store = createVuexStore({
    user: {
      name: "Test",
      email: "test@test.es",
    },
    status: "authenticated",
    idToken: "ABC",
    refreshToken: "XYZ",
  });

  test("should match snapshot", () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  // test("click on logout, you must close the session and redirect", async () => {
  //   const wrapper = shallowMount(Navbar, {
  //     global: {
  //       plugins: [store],
  //       mocks: {
  //         $router: mockRouter,
  //       },
  //     },
  //   });
  //   await wrapper.find("button").trigger("click");
  //
  //   expect(mockRouter.push).toHaveBeenCalled();
  // });
});
