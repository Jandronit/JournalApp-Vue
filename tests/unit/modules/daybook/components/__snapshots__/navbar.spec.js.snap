// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Navbar Component should match snapshot 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white" href=""><img src="" alt="Vue logo" height="24" class="d-inline-block align-text-top mx-2"> Test</a>
  <div class="d-flex"><button class="btn btn-outline-light mx-2"><i class="fas fa-sign-out-alt"></i></button><button class="btn btn-outline-light mx-2">Register</button></div>
</nav>
`;
